### PR TITLE
dev-libs/{libbson,mongo-c-driver}: bump to 1.6.2

### PR DESCRIPTION
**Pull request by hydrapolic**
Closes:
https://bugs.gentoo.org/show_bug.cgi?id=609024
https://bugs.gentoo.org/show_bug.cgi?id=609022

Needed by:
https://bugs.gentoo.org/show_bug.cgi?id=609026

### CODE REVIEW

**Comment by hydrapolic** [comment]
@ultrabug, can you please look at it? Tested on ~amd64 with dev-php/pecl-mongodb-1.2.8.

**Comment by ultrabug** [comment]
I sincerly apologize, I no time to properly get into that.

Any dev or affected user that could validate this feel free to merge & go.

**Comment by mgorny** [comment]
Could you rebase it, please?

**Comment by monsieurp** [comment]
I'll take it. @hydrapolic please rebase and ping me back.

**Comment by hydrapolic** [comment]
@monsieurp, can you please look at it? Thank you!

File: dev-libs/libbson/libbson-1.6.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+src_configure() {
+	econf --disable-optimizations \
+		$(use_enable debug) \
+		$(use_enable static-libs static)
+}
+
+src_install() {
+	default
+
+	if [[ -f "${D}usr/share/doc/${P}/COPYING" ]] ; then
```
Comment (mgorny) [inline dev-libs/libbson/libbson-1.6.2.ebuild]: This looks fairly wrong to me. I suspect the build system should either reliably install or not install the file, and so the removal should be either unconditional or removed altogether. However, it was like this before, so you may ignore this.

File: dev-libs/libbson/libbson-1.6.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		rm "${D}usr/share/doc/${P}/COPYING" || die
+	fi
+
+	# Installing all the manuals conflicts with man-pages
+	doman doc/man/bson_*.3
+	use static-libs || find "${D}" -name '*.la' -delete
```
Comment (mgorny) [inline dev-libs/libbson/libbson-1.6.2.ebuild]: `find` needs `|| die`. So the whole thing needs to be converted into `if ...; then ...; fi`

File: dev-libs/libbson/libbson-1.6.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+	if use examples; then
+		insinto /usr/share/${PF}/examples
```
Comment (mgorny) [inline dev-libs/libbson/libbson-1.6.2.ebuild]: docinto examples
    dodoc ...

File: dev-libs/mongo-c-driver/mongo-c-driver-1.6.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+		--docdir="${EPREFIX}/usr/share/doc/${P}" \
+		$(use_enable sasl) \
+		$(use_enable ssl ssl openssl) \
+		$(use_enable debug) \
+		$(use_enable static-libs static)
+}
+
+src_install() {
+	default_src_install
+	if [[ -f "${D}usr/share/doc/${P}/COPYING" ]] ; then
```
Comment (mgorny) [inline dev-libs/mongo-c-driver/mongo-c-driver-1.6.2.ebuild]: Likewise.

File: dev-libs/mongo-c-driver/mongo-c-driver-1.6.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		rm "${D}usr/share/doc/${P}/COPYING" || die
+	fi
+	# Only install the mongoc man pages to avoid conflicts of common names
+	doman doc/man/mongoc_*.3
+
+	use static-libs || find "${D}" -name '*.la' -delete
```
Comment (mgorny) [inline dev-libs/mongo-c-driver/mongo-c-driver-1.6.2.ebuild]: Likewise.

File: dev-libs/mongo-c-driver/mongo-c-driver-1.6.2.ebuild [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+	if use examples; then
+		insinto /usr/share/${PF}/examples
```
Comment (mgorny) [inline dev-libs/mongo-c-driver/mongo-c-driver-1.6.2.ebuild]: Likewise.

File: dev-libs/mongo-c-driver/mongo-c-driver-1.6.2.ebuild [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+		doins -r examples/*.c examples/aggregation examples/bulk
+	fi
+}
+
+src_test() {
+	# Avoid allocating too much disk space by using server.smallFiles = 1
+	echo -e "storage:\n    smallFiles: true" > "${T}/mongod.conf"
```
Comment (mgorny) [inline dev-libs/mongo-c-driver/mongo-c-driver-1.6.2.ebuild]: ||die

File: dev-libs/mongo-c-driver/mongo-c-driver-1.6.2.ebuild [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+	local PORT=27099
+	mongod --port ${PORT} --bind_ip 127.0.0.1 --nounixsocket --fork \
+		-f "${T}/mongod.conf" --dbpath="${T}" \
+		--logpath="${T}/mongod.log" || die
+	MONGOC_TEST_HOST="127.0.0.1:${PORT}" emake test
+	kill `cat "${T}/mongod.lock"`
```
Comment (mgorny) [inline dev-libs/mongo-c-driver/mongo-c-driver-1.6.2.ebuild]: kill $(<"${T}/mongod.lock")

There's no need to call `cat`.

**Comment by hydrapolic** [comment]
Thanks @mgorny for the suggestions. Just pushed the revised versions.

File: dev-libs/mongo-c-driver/mongo-c-driver-1.6.2.ebuild [hunk 9]
```diff
@@ -0,0 +1,77 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit autotools eutils
```
Comment (mgorny) [inline dev-libs/mongo-c-driver/mongo-c-driver-1.6.2.ebuild]: Is eutils actually used? If not, you don't have to push a fix — just ACK and I'll fix it while merging.

Comment (hydrapolic) [inline dev-libs/mongo-c-driver/mongo-c-driver-1.6.2.ebuild]: You are right, it's not used. Only autotools is used for eautoreconf().

**Comment by hydrapolic** [comment]
Thanks for merging!

**Pull request closed without merging**